include/deal.II/matrix_free/operators.h

```diff
@@ -288,6 +288,48 @@ namespace MatrixFreeOperators
 
 
 
+  /**
+   * This class implements the operation of the action of a mass matrix
+   * and additionally the action of a lumped mass matrix that can be used for
+   * preconditioning. This class is used in VectorTools::project_distributed().
+   *
+   * @author Daniel Arndt, 2016
+   */
+  template <int dim, int fe_degree, int n_components = 1, typename number = double>
+  class MassOperator : public Base<dim, number>
+  {
+  public:
+
+    /**
+     * Constructor. Initializes the MatrixFree object.
+     */
+    MassOperator (const MatrixFree<dim,number> &data);
```

Comment:
Please, make it consistent with the base class, namely use default constructor. Otherwise we don't differentiate between initialization at fine scale (only MatrixFree data is neede) vs initialization at the GMG level (MatrixFree data + MGConstrainedDoFs + level). (see Base class).